File: net/lib/schemas/StringSchema.cs
using Albert.Lib.Rules;

namespace Albert.Lib.Schemas;


public class StringSchema : BaseSchema<string>
{
    public override string Type => "string";

    public StringSchema Min(int min)
    {
        this.AddRule(new StringMinRule(min));
        return this;
    }

    public StringSchema Max(int max)
    {
        this.AddRule(new StringMaxRule(max));
        return this;
    }
}

File: js/lib/schemas/StringSchema.ts
import { StringMaxRule, StringMinRule } from '../rules';
import { BaseSchema } from './BaseSchema';

export class StringSchema extends BaseSchema<string> {
  constructor() {
    super('string');
  }

  min(min: number) {
    return this.addRule(new StringMinRule(min));
  }

  max(max: number) {
    return this.addRule(new StringMaxRule(max));
  }
}
